sky/execution.py

```diff
@@ -630,10 +639,12 @@ def spot_launch(
                         skypilot_config.ENV_VAR_SKYPILOT_CONFIG,
                 })
 
-        yaml_path = backend_utils.fill_template(
-            spot.SPOT_CONTROLLER_TEMPLATE,
-            vars_to_fill,
-            output_prefix=spot.SPOT_CONTROLLER_YAML_PREFIX)
+        yaml_path = os.path.expanduser(
+            os.path.join(spot.SPOT_CONTROLLER_YAML_PREFIX,
+                         f'{name}-{task_uuid}.yaml'))
+        backend_utils.fill_template(spot.SPOT_CONTROLLER_TEMPLATE,
+                                    vars_to_fill,
+                                    output_path=yaml_path)
```

Comment:
output_prefix arg can be removed now, it seems

Reply:
Good point! Removed. Thanks!